src/builder/nix.rs

```diff
@@ -0,0 +1,299 @@
+            }
+        }
+
+        Ok(out_paths)
+    }
+}
+
+#[derive(Clone)]
+pub struct NixBuilder {
+    binary: PathBuf,
+}
+
+impl NixBuilder {
+    async fn eval(
+        &self,
+        mut progress: Item,
+        set: &mut JoinSet<Result<OutPaths, NixError>>,
+        service_name: &str,
+        platform: &str,
+        platforms: &[String],
+        packages: &HashMap<String, String>,
+    ) -> Result<(), NixError> {
+        let gc_roots_dir = std::env::temp_dir();
+
+        let mut root_cmd = Command::new(&self.binary);
+        let mut cmd = root_cmd
+            .arg("--verbose")
+            .arg("--log-format")
+            .arg("internal-json")
+            .arg("--gc-roots-dir")
+            .arg(&gc_roots_dir);
+
+        let system = try_system(platform)?;
+
+        match platforms.iter().find(|p| p == &&system) {
+            Some(platform) => {
+                cmd = cmd.arg("--flake").arg(format!(".#packages.{platform}"));
+                progress.message(MessageLevel::Info, format!("using platform: {platform}"));
+            }
+            None => {
+                return Ok(());
+            }
+        }
+
+        let child = exec::spawn(cmd).await?;
+
+        progress::proxy_stdio(child.stderr, progress.add_child("nix").into());
+
+        let reader = BufReader::new(child.stdout);
+        let mut lines = reader.lines();
+
+        while let Some(line) = lines.next_line().await? {
+            let drv: EvalResult = serde_json::from_str(&line)?;
+
+            if packages.iter().any(|(_, v)| v == &drv.attr_path.join(".")) {
+                set.spawn(drv.build(progress.add_child(format!("{service_name} › nix"))));
+            }
+        }
+
+        Ok(())
+    }
+}
+
+impl Builder for NixBuilder {
+    type Error = NixError;
+    type Input = Nix;
+
+    fn try_init() -> Result<Self, Self::Error> {
+        which::which("nix-eval-jobs")
+            .map_err(Self::Error::from)
+            .map(|binary| Self { binary })
+    }
+
+    async fn build(
+        self,
+        mut progress: prodash::tree::Item,
+        Context {
+            service_name,
+            platform,
+            input,
+        }: Context<Self::Input>,
+    ) -> Result<Output, Self::Error> {
+        progress.set_name(&service_name);
+        progress.message(MessageLevel::Info, "starting builder");
+
+        let mut set = JoinSet::default();
+        self.eval(
+            progress.add_child("eval"),
+            &mut set,
+            &service_name,
+            &platform,
+            &input.platforms,
+            &input.packages,
+        )
+        .await?;
+
+        progress.message(MessageLevel::Success, "evaluation finished".to_string());
+
+        let out_paths =
+            set.join_all()
+                .await
+                .into_iter()
+                .try_fold(OutPaths::new(), |mut acc, paths| {
+                    acc.extend(paths?);
+                    Ok::<_, NixError>(acc)
+                })?;
```

Comment:
It's optional but I think it would be nice to show sub-progress here and use `progress.init(..)`/`progress.inc()`